module-system/sov-cli/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+use std::env;
+use std::path::{Path, PathBuf};
+
+use directories::BaseDirs;
+pub use sov_modules_api::clap;
+
+pub mod wallet_state;
+pub mod workflows;
+
+const SOV_WALLET_DIR_ENV_VAR: &str = "SOV_WALLET_DIR";
+
+/// The directory where the wallet is stored.
+pub fn wallet_dir() -> Result<impl AsRef<Path>, anyhow::Error> {
+    // First try to parse from the env variable
+    if let Ok(val) = env::var(SOV_WALLET_DIR_ENV_VAR) {
+        return PathBuf::try_from(val).map_err(|e: std::convert::Infallible| {
```

Comment:
Why do we `map_err` on `std::convert::Infallible` since it can never happen?

Reply:
Oops! Haha I let my IDE fill in that type annotation. Switching to `val.into()`.